sentry-core/src/main/java/io/sentry/core/Sentry.java

```diff
@@ -160,9 +154,65 @@ public static void init(
     }
   }
 
+  private static boolean initConfigurations(final @NotNull SentryOptions options) {
+    final String dsn = options.getDsn();
+    if (dsn == null) {
+      throw new IllegalArgumentException("DSN is required. Use empty string to disable SDK.");
+    } else if (dsn.isEmpty()) {
+      close();
+      return false;
+    }
+
+    @SuppressWarnings("unused")
+    final Dsn parsedDsn = new Dsn(dsn);
+
+    ILogger logger = options.getLogger();
+
+    // check if its no op, might be agnostic?
+    if (options.isDebug() && logger instanceof NoOpLogger) {
+      options.setLogger(new SystemOutLogger());
+      logger = options.getLogger();
+    }
+    logger.log(SentryLevel.INFO, "Initializing SDK with DSN: '%s'", options.getDsn());
+
+    if (options.getSentryClientName() == null || options.getSentryClientName().isEmpty()) {
+      options.setSentryClientName("sentry.java/2.2.0"); // TODO: get from build script or CI envs
+    }
+
+    // TODO: read values from conf file eg Manifest
+    // eg release and distinctId
+
+    if (options.getEnvelopeReader() instanceof NoOpEnvelopeReader) {
```

Comment:
What about we jsut add `EnvelopeReader` by default instead of `NoOpEnvelopeReader`?

Reply:
wfm, actually I even wanted to do this with other objects that should not be replaced anyway, we can discuss what it's make sense to be the object by default and not the NoOp version of it

Reply:
Fair. Let's aim at: If it's likely the default implementation will be used (or we most certainly need one that is not NoOp) we should initialize with our default instance.